@here/harp-mapview/lib/text/TextElementStateCache.ts

```diff
@@ -115,61 +154,61 @@ export class TextElementStateCache {
      * @returns True if it's the remaining element after deduplication, false if it's been marked
      * as duplicate.
      */
-    deduplicateElement(elementState: TextElementState): boolean {
+    deduplicateElement(elementState: TextElementState, zoomLevel: number): boolean {
         const element = elementState.element;
 
         if (element.type !== TextElementType.PoiLabel) {
             return true;
         }
 
-        // Point labels may have duplicates (as can path labels), Identify them
-        // and keep the one we already display.
-
-        const cachedEntries = this.m_textMap.get(element.text);
+        const cacheResult = this.findDuplicate(elementState, zoomLevel);
 
-        if (cachedEntries === undefined) {
-            // No labels found with the same text.
+        if (cacheResult === undefined) {
+            // Text not found so far, add this element to cache.
             this.m_textMap.set(element.text, [elementState]);
             return true;
         }
 
-        // Other labels found with the same text. Check if they're near enough to be considered
-        // duplicates.
-        const entryCount = cachedEntries.length;
-        const elementPosition = element.points as THREE.Vector3;
-        let duplicateIndex: number = -1;
-        for (let i = 0; i < entryCount; ++i) {
-            const cachedElementPosition = cachedEntries[i].element.points as THREE.Vector3;
-
-            if (
-                elementPosition.distanceToSquared(cachedElementPosition) <
-                MAX_LABEL_DUPLICATE_DIST_SQ
-            ) {
-                duplicateIndex = i;
-                break;
-            }
-        }
-
-        if (duplicateIndex === -1) {
-            // No duplicate found.
-            cachedEntries.push(elementState);
+        if (cacheResult.index === -1) {
+            // No duplicate found among elements with same text,add this one to cache.
+            cacheResult.entry.push(elementState);
             return true;
         }
 
         // Duplicate found, check whether there's a label already visible and keep that one.
-        const cachedDuplicate = cachedEntries[duplicateIndex];
+        const cachedDuplicate = cacheResult.entry[cacheResult.index];
 
         if (!cachedDuplicate.visible && elementState.visible) {
             // New label is visible, substitute the cached label.
-            cachedEntries[duplicateIndex] = elementState;
-
+            cacheResult.entry[cacheResult.index] = elementState;
             cachedDuplicate.reset();
             return true;
         }
 
         return false;
     }
 
+    replaceElement(zoomLevel: number, elementState: TextElementState): void {
```

Comment:
please add some documentation to this function.

Reply:
done